experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -42,6 +82,41 @@ export async function useAzureSocketIOChain(
   return this;
 }
 
+function getNegotiateHandler(): (
+  req: IncomingMessage,
+  res: ServerResponse,
+  configureNegotiateOptions: (req: IncomingMessage) => Promise<NegotiateOptions>,
+  serviceClient: WebPubSubServiceClient
+) => Promise<void> {
+  return async (
+    req: IncomingMessage,
+    res: ServerResponse,
+    configureNegotiateOptions: (req: IncomingMessage) => Promise<NegotiateOptions>,
+    serviceClient: WebPubSubServiceClient
+  ): Promise<void> => {
+    let statusCode = 500;
+    let message = {};
+    try {
+      const negotiateOptions = await configureNegotiateOptions(req);
+
+      // Example: https://<web-pubsub-endpoint>?access_token=ABC.EFG.HIJ
+      const tokenResponse = await serviceClient.getClientAccessToken(negotiateOptions);
+      const url = new URL(req.url, tokenResponse.baseUrl);
+      const protocol = url.protocol.replace("wss", "https").replace("ws", "http");
+      const endpointWithToken = `${protocol}//${url.host}?access_token=${tokenResponse.token}`;
+
+      statusCode = 200;
+      message = { endpoint: endpointWithToken };
+    } catch (e) {
+      statusCode = 500;
+      message = { message: e.message };
```

Comment:
I am not sure if server should expose the error message to the client, maybe log error detail and message to a general "Internal server error"

Reply:
updated as suggested